jbpm-task-assigning/jbpm-task-assigning-core/src/test/java/org/jbpm/task/assigning/model/solver/realtime/AbstractProblemFactChangeTest.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.task.assigning.model.solver.realtime;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Function;
+
+import org.jbpm.task.assigning.AbstractTaskAssigningCoreTest;
+import org.jbpm.task.assigning.model.TaskAssigningSolution;
+import org.junit.Rule;
+import org.junit.rules.ExpectedException;
+import org.optaplanner.core.api.solver.Solver;
+import org.optaplanner.core.impl.solver.ProblemFactChange;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public abstract class AbstractProblemFactChangeTest extends AbstractTaskAssigningCoreTest {
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    final Random random = new Random();
```

Comment:
This `Random` instance seems to be unused.

Reply:
as discussed it's used in subclasses